Makefile

```diff
@@ -217,22 +225,29 @@ rm -rf $$TMP_DIR ;\
 endef
 
 .PHONY: bundle
-bundle: manifests kustomize operator-sdk ## Generate bundle manifests and metadata, then validate generated files.
+bundle: manifests kustomize operator-sdk ko ## Generate bundle manifests and metadata, then validate generated files.
 	$(OPERATOR_SDK) generate kustomize manifests --interactive=false -q
-	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	$(OPERATOR_SDK) bundle validate ./bundle
 
 .PHONY: verify-bundle
 verify-bundle: bundle ## Verify bundle manifests were generated and committed to git
 	hack/check-git-status.sh bundle
 
+OPERATOR_CSV = bundle/manifests/shipwright-operator.clusterserviceversion.yaml
+
 .PHONY: bundle-build
-bundle-build: bundle ## Build the bundle image.
-	$(CONTAINER_ENGINE) build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+bundle-build: bundle ko ## Build the bundle image. If IMAGE_PUSH is set to true, this will also build and push the operator image.
+	rm -rf _output/olm
+	mkdir -p _output/olm
+	cp -r bundle _output/olm/
+	cp bundle.Dockerfile _output/olm/
+	KO_DOCKER_REPO=${IMAGE_REPO} $(KO) resolve --push=${IMAGE_PUSH} ${KO_OPTS} -f ${OPERATOR_CSV} > _output/olm/${OPERATOR_CSV}
+	$(CONTAINER_ENGINE) build -f _output/olm/bundle.Dockerfile -t $(BUNDLE_IMG) _output/olm
```

Comment:
Sad that OLM undoes ko's hard work and requires a container runtime to make this bundle. Without knowing much about it I'd guess it's entirely unnecessary.

Reply:
It doesn't undo ko's hard work, per se. The bundle is just YAML in an OCI artifact, which are updated with the `ko resolve`d image shas. A full runtime like docker/podman is indeed overkill, but in terms of what developers reasonably have available, the decision makes sense.